File: postman/sdks/Collections SDKs/api/typescript/src/http/environment.ts
/**
 * Available API environments with their base URLs.
 * Use these constants to configure the SDK for different environments (production, staging, etc.).
 */
export enum Environment {
  /** DEFAULT environment base URL */
  DEFAULT = 'https://{{QAUrl}}api',
  /** QAURLAPI environment base URL */
  QAURLAPI = 'https://{{QAUrl}}api',
  /** QAURLAPI_1 environment base URL */
  QAURLAPI_1 = 'https://{{QAUrl}}API',
}

File: postman/sdks/Collections SDKs/api/java/src/main/java/com/apidemocollectionsdk/http/Environment.java
package com.apidemocollectionsdk.http;

import lombok.Getter;
import okhttp3.HttpUrl;

/**
 * Predefined environment configurations for the SDK.
 * Each environment represents a different base URL (e.g., production, staging, development).
 * URLs are validated at construction time to ensure they are well-formed.
 */
@Getter
public enum Environment {
  DEFAULT("https://{{QAUrl}}api"),
  QAURLAPI("https://{{QAUrl}}api"),
  QAURLAPI_1("https://{{QAUrl}}API");

  private final String url;

  /**
   * Creates an environment with the specified base URL.
   * Validates that the URL is well-formed before accepting it.
   *
   * @param url The base URL for this environment
   * @throws IllegalArgumentException if the URL is invalid
   */
  Environment(String url) {
    if (HttpUrl.parse(url) == null) {
      throw new IllegalArgumentException(
        String.format(
          "Environment url [%s] is not valid. Please use the following format https://api.example.com",
          url
        )
      );
    }
    this.url = url;
  }
}
